### PR TITLE
ES: Add El Futuro en tu mano + fix streams

### PR DESCRIPTION
**Pull request by joaquinito2048**

### CODE REVIEW

**Comment by LaneSh4d0w** [comment]
Do you appear to have any alternative links for the 7TV links you've just removed ?
I'm glad btw that someone forked my query-streamlink repo and is willing to use the host for here =)
You may add the free.fullspeed host from @dct-infra as an alternate link to your host.

**Comment by joaquinito2048** [comment]
@LaneSh4d0w 7TV is not working.

**Comment by joaquinito2048** [comment]
@LaneSh4d0w So, i don't have any alternative links.

**Comment by joaquinito2048** [comment]
@LaneSh4d0w I are using my own query-streamlink in Heroku.

**Comment by LaneSh4d0w** [comment]
> @LaneSh4d0w 7TV is not working.

I see. Maybe putting them as Offline in case they come back would have been better.

**Comment by joaquinito2048** [comment]
@LaneSh4d0w Is not working anymore. I remove because 7TV Andalucía does not work anymore.

File: channels/es.m3u [hunk 1]
```diff
@@ -101,16 +79,12 @@ https://5924d3ad0efcf.streamlock.net/canal4/canal4live/playlist.m3u8
...
-#EXTINF:-1 tvg-id="Canal24horas.es" tvg-country="ES" tvg-language="Catalan" tvg-logo="" group-title="",Canal 24 horas (1080p) [Timeout]
+#EXTINF:-1 tvg-id="Canal24horas.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://upload.wikimedia.org/wikipedia/commons/thumb/4/4c/Logo_TVE-24h.svg/2084px-Logo_TVE-24h.svg.png" group-title="News",Canal 24 horas (1080p)
 https://rtvelivestreamv3.akamaized.net/24h_bkp_dvr.m3u8
-#EXTINF:-1 tvg-id="Canal24horas.es" tvg-country="ES" tvg-language="Catalan" tvg-logo="" group-title="",Canal 24 horas (720p) [Timeout]
+#EXTINF:-1 tvg-id="Canal24horas.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://upload.wikimedia.org/wikipedia/commons/thumb/4/4c/Logo_TVE-24h.svg/2084px-Logo_TVE-24h.svg.png" group-title="News",Canal 24 horas (720p)
 https://rtvelivestreamv3.akamaized.net/24h_main_dvr.m3u8
+#EXTINF:-1 tvg-id="Canal24horas.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://upload.wikimedia.org/wikipedia/commons/thumb/4/4c/Logo_TVE-24h.svg/2084px-Logo_TVE-24h.svg.png" group-title="News",Canal 24 horas (1080p)
+https://streamlink-iptv.herokuapp.com/iptv-query?streaming-ip=https://www.youtube.com/watch?v=_BOSGp6dzas
```
Comment (LaneSh4d0w) [inline channels/es.m3u:87]:
```suggestion
https://streamlink-iptv.herokuapp.com/iptv-query?streaming-ip=https://www.youtube.com/channel/UC7QZIf0dta-XPXsp9Hv4dTw/live
```

File: channels/es.m3u [hunk 2]
```diff
@@ -141,16 +115,16 @@ http://217.125.136.93:8080/canalsierradecadiz720.m3u8
 http://217.125.136.93:8080/canalsierradecadiz576.m3u8
 #EXTINF:-1 tvg-id="CanalSuper3.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://i.imgur.com/kbgoyvg.png" group-title="",Canal Super 3 (1080p) [Not 24/7]
 https://directes-tv-int.ccma.cat/int/ngrp:c33_web/playlist.m3u8
-#EXTINF:-1 tvg-id="CanalSurAndalucia.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://i.imgur.com/sEN6VcY.png" group-title="",Canal Sur Andalucía (720p) [Not 24/7]
-https://cdnlive.shooowit.net/rtvalive/smil:83CRxkDnV6DVR.smil/playlist.m3u8
+#EXTINF:-1 tvg-id="CanalSurAndalucia.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://i.imgur.com/sEN6VcY.png" group-title="",Canal Sur Andalucía (1080p)
+https://streamlink-iptv.herokuapp.com/iptv-query?streaming-ip=https://www.youtube.com/watch?v=TpV-x9ICNvo
```
Comment (LaneSh4d0w) [inline channels/es.m3u:119]:
```suggestion
https://streamlink-iptv.herokuapp.com/iptv-query?streaming-ip=https://www.youtube.com/channel/UCqwjtgaJflHga2aPWphG5PQ/live
```

File: channels/es.m3u [hunk 3]
```diff
@@ -189,6 +163,10 @@ https://etbvnogeo-lh.akamaihd.net/i/ETBEITBEUS_1@300391/master.m3u8
 https://etbvnogeo-lh.akamaihd.net/i/ETBSTR2_1@595582/master.m3u8
 #EXTINF:-1 tvg-id="El33.es" tvg-country="ES;AD" tvg-language="Spanish" tvg-logo="https://graph.facebook.com/Canal33/picture?width=320&height=320" group-title="",El 33 (1080p) [Not 24/7]
 https://directes-tv-int.ccma.cat/int/ngrp:c33_web/playlist_DVR.m3u8
+#EXTINF:-1 tvg-id="ElFuturoentumano.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="" group-title="Religious",El Futuro en tu mano (720p)
+https://limited24.todostreaming.es/live/renjillo-livestream.m3u8
+#EXTINF:-1 tvg-id="ElFuturoentumano.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="" group-title="Religious",El Futuro en tu mano (720p)
+https://streamlink-iptv.herokuapp.com/iptv-query?streaming-ip=https://www.youtube.com/watch?v=4_hoC_Nzu0Q
```
Comment (LaneSh4d0w) [inline channels/es.m3u:169]:
```suggestion
https://streamlink-iptv.herokuapp.com/iptv-query?streaming-ip=https://www.youtube.com/channel/UCZSw3jfFZo_8I9BHSMlT2Fg/live
```

File: channels/es.m3u [hunk 4]
```diff
@@ -265,6 +243,8 @@ https://livelist01.yowi.tv/lista/0d6c7ccfac89946bfd41ae34c527e8d94734065c/master
 https://livelist01.yowi.tv/lista/8635ae40f8d1a32eccd63d1f58b55662c9c98f9f/master.m3u8
 #EXTINF:-1 tvg-id="HuescaTV.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://i.imgur.com/hmYDqGV.png" group-title="",Huesca TV (240p) [Not 24/7]
 https://streaming2.radiohuesca.com/hls-live/livepkgr/_definst_/huescatv/huescatv.m3u8
+#EXTINF:-1 tvg-id="IB3Global.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="" group-title="",IB3 Global (720p)
+https://streamlink-iptv.herokuapp.com/iptv-query?streaming-ip=https://www.youtube.com/watch?v=BxmVws5iYB4
```
Comment (LaneSh4d0w) [inline channels/es.m3u:247]:
```suggestion
https://streamlink-iptv.herokuapp.com/iptv-query?streaming-ip=https://www.youtube.com/channel/UCff_CBVJDTHP4wOHPjP5BMg/live
```

**Review by BellezaEmporium — CHANGES_REQUESTED** [review]
see comments for changes to make.

File: channels/es.m3u [hunk 5]
```diff
@@ -345,8 +349,8 @@ https://mdslivehlsb-i.akamaihd.net/hls/live/875601/niusdiario/03.m3u8
 https://mdslivehlsb-i.akamaihd.net/hls/live/875601-b/niusdiario/01.m3u8
 #EXTINF:-1 tvg-id="NoroesteTV.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="" group-title="",Noroeste TV (720p) [Not 24/7]
 https://cloudvideo.servers10.com:8081/8032/index.m3u8
-#EXTINF:-1 tvg-id="NORTEvision.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="" group-title="",NORTEvisión [Offline]
-https://query-streamlink.herokuapp.com/iptv-query?streaming-ip=https://facebook.com/1918671788399481/videos/3060444244223429/
+#EXTINF:-1 tvg-id="NORTEvision.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="" group-title="",NORTEvisión
+http://free.fullspeed.tv/iptv-query?streaming-ip=https://facebook.com/1918671788399481/videos/3060444244223429/
```
Comment (LaneSh4d0w) [inline channels/es.m3u:353]: Not quite sure this is the correct FB stream. It leads me to a 18 minute-long video.

Comment (joaquinito2048) [reply channels/es.m3u]: Fixed.

File: channels/es.m3u [hunk 6]
```diff
@@ -101,16 +79,12 @@ https://5924d3ad0efcf.streamlock.net/canal4/canal4live/playlist.m3u8
...
+#EXTINF:-1 tvg-id="Canal24horas.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://upload.wikimedia.org/wikipedia/commons/thumb/4/4c/Logo_TVE-24h.svg/2084px-Logo_TVE-24h.svg.png" group-title="News",Canal 24 horas (1080p)
 https://rtvelivestreamv3.akamaized.net/24h_bkp_dvr.m3u8
-#EXTINF:-1 tvg-id="Canal24horas.es" tvg-country="ES" tvg-language="Catalan" tvg-logo="" group-title="",Canal 24 horas (720p) [Timeout]
+#EXTINF:-1 tvg-id="Canal24horas.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://upload.wikimedia.org/wikipedia/commons/thumb/4/4c/Logo_TVE-24h.svg/2084px-Logo_TVE-24h.svg.png" group-title="News",Canal 24 horas (720p)
 https://rtvelivestreamv3.akamaized.net/24h_main_dvr.m3u8
+#EXTINF:-1 tvg-id="Canal24horas.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://upload.wikimedia.org/wikipedia/commons/thumb/4/4c/Logo_TVE-24h.svg/2084px-Logo_TVE-24h.svg.png" group-title="News",Canal 24 horas (1080p)
+https://streamlink-iptv.herokuapp.com/iptv-query?streaming-ip=https://www.youtube.com/channel/UC7QZIf0dta-XPXsp9Hv4dTw/live
```
Comment (LaneSh4d0w) [inline channels/es.m3u:87]:
```suggestion
https://streamlink-iptv.herokuapp.com/iptv-query?streaming-ip=https://www.youtube.com/channel/UC7QZIf0dta-XPXsp9Hv4dTw/live
#EXTINF:-1 tvg-id="Canal24horas.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://upload.wikimedia.org/wikipedia/commons/thumb/4/4c/Logo_TVE-24h.svg/2084px-Logo_TVE-24h.svg.png" group-title="News",Canal 24 horas (1080p)
http://free.fullspeed.tv/iptv-query?streaming-ip=https://www.youtube.com/channel/UC7QZIf0dta-XPXsp9Hv4dTw/live
```
For the time being, add the other one as alternative. I know your Heroku follows my release, and @dct-infra managed to put the latest master of Streamlink, making it work.

**Comment by joaquinito2048** [comment]
@LaneSh4d0w Removed my server because it's not working. Instead, will use @dct-infra server.

**Review by BellezaEmporium — DISMISSED** [review]

**Comment by joaquinito2048** [comment]
@Dum4G Please, approve these changes and merge it now.

**Comment by Dum4G** [comment]
@freearhey should we preserve a group of channels that went offline for unknown period? Or is it gonna be removed automatically anyways?

**Comment by freearhey** [comment]
@Dum4G Broken links are stored in playlists to save channel descriptions, and the script only removes duplicate links having the same tvg-id.

File: channels/es.m3u [hunk 7]
```diff
@@ -27,34 +27,12 @@ https://pc-la7delarioja-hls.flumotion.com/playlist.m3u8
 https://amg01573-7nn-7nnono-ono-pcdj3.amagi.tv/playlist.m3u8
 #EXTINF:-1 tvg-id="7TelevisionRegiondeMurcia.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://i.imgur.com/M0JqHa9.png" group-title="Local",7 Televisión Región de Murcia (360p)
 https://rtvmurcia_01-lh.akamaihd.net/i/rtvmurcia_1_0@507973/master.m3u8
```
Comment (Dum4G) [inline channels/es.m3u:29]:
```suggestion
https://rtvmurcia_01-lh.akamaihd.net/i/rtvmurcia_1_0@507973/master.m3u8
#EXTINF:-1 tvg-id="7TVAndalucia.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://graph.facebook.com/7TelevisionAndalucia/picture?width=320&height=320" group-title="News",7 TV Andalucía (720p) [Offline]
https://dcunilive93-lh.akamaihd.net/i/dclive_1@835804/master.m3u8
#EXTINF:-1 tvg-id="7TVAndaluciaBahia.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://graph.facebook.com/7TelevisionAndalucia/picture?width=320&height=320" group-title="News",7 TV Andalucía (Bahía) (576p) [Offline]
https://dcunilive93-lh.akamaihd.net/i/dclive_1@835790/master.m3u8
#EXTINF:-1 tvg-id="7TVAndaluciaCordoba.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://graph.facebook.com/7TelevisionAndalucia/picture?width=320&height=320" group-title="News",7 TV Andalucía (Córdoba) (720p) [Not 24/7]
https://dcunilive265-lh.akamaihd.net/i/dclive_1@409360/master.m3u8
#EXTINF:-1 tvg-id="7TVAndaluciaCostaNoroeste.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://graph.facebook.com/7TelevisionAndalucia/picture?width=320&height=320" group-title="News",7 TV Andalucía (Costa Noroeste) (404p) [Offline]
https://dcunilive93-lh.akamaihd.net/i/dclive_1@835802/master.m3u8
#EXTINF:-1 tvg-id="7TVAndaluciaJaen.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://graph.facebook.com/7TelevisionAndalucia/picture?width=320&height=320" group-title="News",7 TV Andalucía (Jaén) (404p) [Offline]
https://dcunilive266-lh.akamaihd.net/i/dclive_1@426886/master.m3u8
#EXTINF:-1 tvg-id="7TVAndaluciaJerez.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://graph.facebook.com/7TelevisionAndalucia/picture?width=320&height=320" group-title="News",7 TV Andalucía (Jerez) (404p) [Offline]
https://dcunilive93-lh.akamaihd.net/i/dclive_1@835794/master.m3u8
#EXTINF:-1 tvg-id="7TVAndaluciaMalaga.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://graph.facebook.com/7TelevisionAndalucia/picture?width=320&height=320" group-title="News",7 TV Andalucía (Málaga) [Offline]
https://dcunilive93-lh.akamaihd.net/i/dclive_1@835798/master.m3u8
#EXTINF:-1 tvg-id="7TVAndaluciaSevilla.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://graph.facebook.com/7TelevisionAndalucia/picture?width=320&height=320" group-title="News",7 TV Andalucía (Sevilla) (404p) [Offline]
https://dcunilive93-lh.akamaihd.net/i/dclive_1@835800/master.m3u8
#EXTINF:-1 tvg-id="7TVAndaluciaSierra.es" tvg-country="ES" tvg-language="Spanish" tvg-logo="https://graph.facebook.com/7TelevisionAndalucia/picture?width=320&height=320" group-title="News",7 TV Andalucía (Sierra) (404p) [Offline]
https://dcunilive93-lh.akamaihd.net/i/dclive_1@835792/master.m3u8
```

**Review by Dum4G — APPROVED** [review]

**Review by BellezaEmporium — APPROVED** [review]

**Comment by Dum4G** [comment]
Thanks for contribution!

**Pull request merged**